compiler/src/iree/compiler/Codegen/LLVMGPU/Passes.cpp

```diff
@@ -197,6 +197,36 @@ void addGPUMatmulTensorCorePassPipeline(OpPassManager &pm,
       createGPUPipeliningPass(pipelineDepth));
 }
 
+void addGPUTransposePassPipeline(OpPassManager &pm) {
+  llvm::dbgs() << "LLVMGPU::addGPUTransposePassPipeline()\n";
+  tileAndBufferize(pm);
+
+  auto &nestedModulePM = pm.nest<ModuleOp>();
+  // Distribute linalg onto threads within the workgroup.
+  nestedModulePM.addNestedPass<func::FuncOp>(createLLVMGPUTileAndDistribute(
+      false, GPUPromoteSharedMemPattern::TransposeOpPattern));
+  nestedModulePM.addNestedPass<func::FuncOp>(createMemrefCopyToLinalgPass());
+  nestedModulePM.addNestedPass<func::FuncOp>(
+      createGPUDistributeSharedMemoryCopy());
+  nestedModulePM.addPass(createCanonicalizerPass());
+  nestedModulePM.addPass(createCSEPass());
+
+  // May or may not need to reduce shared mememory conflicts
+  nestedModulePM.addNestedPass<func::FuncOp>(
+      createLLVMGPUReduceSharedMemoryBankConflicts());
```

Comment:
This transformation is pretty much what you want for bank conflicts reduction but this pass is right now hardcoded to handle the matmul case, you'll need to pass a parameter to do what you want. You can either skip this pass for now or add the parameter in this change

Reply:
I took a look at this pass but I cannot figure out what parameter you're referring to. Looking at the IR it does look like it is activating though. It pads the following alloc: 

```mlir
  %3 = memref.alloc() : memref<32x32xf32, 3>
```

to this:

```mlir
%3 = memref.alloc() : memref<32x36xf32, 3>
  %4 = memref.subview %3[0, 0] [32, 32] [1, 1] : memref<32x36xf32, 3> to memref<32x32xf32, affine_map<(d0, d1) -> (d0 * 36 + d1)>, 3>
  
```

Reply:
exactly, and how much you are padding the alloc depends on the memory accesses you are expecting. Here it is padding with 16bytes but that's most likely not what you want for you case

Reply:
Updated the `LLVMGPUReduceBankConflicts` to accept padding size. Now padding to 33 bytes instead of 36.